core/src/main/java/org/elasticsearch/indices/recovery/RecoveryFailedException.java

```diff
@@ -46,7 +46,8 @@ public RecoveryFailedException(ShardId shardId, DiscoveryNode sourceNode, Discov
     }
 
     public RecoveryFailedException(ShardId shardId, DiscoveryNode sourceNode, DiscoveryNode targetNode, @Nullable String extraInfo, Throwable cause) {
-        super(shardId + ": Recovery failed from " + sourceNode + " into " + targetNode + (extraInfo == null ? "" : " (" + extraInfo + ")"), cause);
+        super(shardId + ": Recovery failed" + (sourceNode != null ? " from " + sourceNode : "") +
```

Comment:
I'm wondering if we should adapt the whole message and say "Recovery failed on " + targetNode , if there is no sourceNode